### PR TITLE
Fix scrollbar error message and test

### LINKED ISSUE #84569: Incorrect logic in the Scrollbar error message
After merging https://github.com/flutter/flutter/pull/81278 I've noticed that the following `if` is actually reversed:
https://github.com/flutter/flutter/blob/f1542ace4fd8ca2de78e49f4bb0def4849e1f40c/packages/flutter/lib/src/widgets/scrollbar.dart#L1050

On top of that, a related test in the framework is not properly checking the exceptions that the above code throws.
The `catch` clauses are not getting called, so in reality, nothing is being tested here: https://github.com/flutter/flutter/blob/f1542ace4fd8ca2de78e49f4bb0def4849e1f40c/packages/flutter/test/material/scrollbar_test.dart#L1438

### PR DESCRIPTION
**Pull request by davidmartos96**
This PR fixes some if/else logic in the creation of the Scrollbar error messages and improves an existing test that wasn't testing properly that code. More information in the linked issue.

Fixes: https://github.com/flutter/flutter/issues/84569

@Piinks That if/else and original test got introduced in this PR https://github.com/flutter/flutter/pull/77107 

On the same note, while testing the merged code in https://github.com/flutter/flutter/pull/81278 I think we could make an even better error message if it actually told which Scrollbar is throwing the error. RIght now the error shows as follows, but it's impossible to tell which scrollbar is incorrect. This could catch off guard people with a big codebase when this gets into stable.
image

I've asked in the hackers Discord what would be the best way to tackle this. 
https://discord.com/channels/608014603317936148/608021234516754444/853999114018750484

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Hey @davidmartos96! Thank you for the contribution!

File: packages/flutter/test/material/scrollbar_test.dart [hunk 1]
```diff
@@ -1448,8 +1448,12 @@ void main() {
       );
     }
 
-    Widget _buildApp({ ScrollController? scrollController }) {
+    Widget _buildApp({
+      required String id,
+      ScrollController? scrollController,
+    }) {
       return MaterialApp(
+        key: ValueKey<String>(id),
```
[the remark this answers is not in the document]
Comment (davidmartos96) [reply packages/flutter/test/material/scrollbar_test.dart:1456]: Yes, while improving the test I noticed that the second call of `pumpWidget(buildApp))` maintains the state of the dragged TabBarView,  so the key effectively resets the state of the whole app for the new drag gesture

[the remark this answers is not in the document]
Comment (Piinks) [reply packages/flutter/test/material/scrollbar_test.dart:1456]: Ah! Thank you. Fantastic. :)

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Pull request merged**